### DISCUSSION TITLE
Can I make my babel plugin execute before the internal vite:react-babel?

### DISCUSSION THREAD

**Discussion by libondev** [vitejs/vite-plugin-react#399] [Q&A] [answered]
I want to implement a plug-in, which is similar to the function of v-model, but my babel plug-in found that the code had been tree-shake after getting the code, and when I only passed in the `count`, the `setCount` had been deleted, so this problem arose.
![image](https://github.com/user-attachments/assets/49c5bea9-4a46-4559-afbc-0e33a12ed1be)

**Comment by ArnaudBarre** [comment, marked as answer]
From the output it looks like you are using the react compiler. I don't know what exactly what your v-model syntax does, but surely it will breaks assumptions from the compiler. Plus it will also breaks various lint rules. I really don't recommend to break out of React and the ecosystem for a small syntax change. Maybe you prefer Vue to React, but trying to do vue with React will just make things worth than raw React.
